lisa/executable.py

```diff
@@ -20,6 +21,11 @@
 T = TypeVar("T")
 
 
+# circular dep if we use the helper in node.py, unfortunately
+def _is_node_remote(node: Node) -> bool:
```

Comment:
How about define `is_remote` to a member of `Node`? It will not be cycle reference.

Reply:
I'd rather not have a method on the parent about children

Reply:
See my PR, it can be an abstract method, no type checking.

Reply:
It's still ugly to infer anything about extensions on a base class, that breaks OO totally :/

Reply:
`is_remote` is a feature on node like `is_posfix`. Some case cannot run locally, like smoke_test. It's high cost to reboot current machine, and resume tests.

Reply:
I know, but is belongs in the module, not the class. OS is one class only, node is not anymore. The module knows there is more to a Node, the class should not have anything to do with it.